### PR TITLE
Grammar cleanup

### PR DESCRIPTION
**Pull request by blueedgenick**
Remove unused terms from grammar and AST code. 

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4 [hunk 1]
```diff
@@ -265,36 +256,25 @@ valueExpression
...
     : NULL                                                                           #nullLiteral
-    | interval                                                                       #intervalLiteral
     | identifier STRING                                                              #typeConstructor
     | number                                                                         #numericLiteral
     | booleanValue                                                                   #booleanLiteral
     | STRING                                                                         #stringLiteral
     | BINARY_LITERAL                                                                 #binaryLiteral
-    | POSITION '(' valueExpression IN valueExpression ')'                            #position
-    | qualifiedName '(' ASTERISK ')' over?                                           #functionCall
-    | qualifiedName '(' (expression (',' expression)*)? ')' over?                    #functionCall
```
Comment (hjafarpour) [inline ksql-parser/src/main/antlr4/io/confluent/ksql/parser/SqlBase.g4]: We do need this for our function calls!

File: ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java [hunk 2]
```diff
@@ -1291,40 +1184,6 @@ public Node visitWhenClause(final SqlBaseParser.WhenClauseContext context) {
     );
   }
 
-  @Override
-  public Node visitFunctionCall(final SqlBaseParser.FunctionCallContext context) {
-    final Optional<Window> window = visitIfPresent(context.over(), Window.class);
```
Comment (hjafarpour) [inline ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: This one also is needed for functionCalls.

File: ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java [hunk 3]
```diff
@@ -1118,7 +1072,7 @@ public Node visitParenthesizedExpression(
 
   @Override
   public Node visitCast(final SqlBaseParser.CastContext context) {
-    final boolean isTryCast = context.TRY_CAST() != null;
+    final boolean isTryCast = false;
```
Comment (hjafarpour) [inline ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: Shall we get rid of `isTryCast`? It won't be used anyway.

Comment (blueedgenick) [reply ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: yeah, i think that makes sense, not sure we'll ever want to use this. Removing more code is good!

File: ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java [hunk 4]
```diff
@@ -1005,10 +972,7 @@ public Node visitNullPredicate(final SqlBaseParser.NullPredicateContext context)
 
   @Override
   public Node visitLike(final SqlBaseParser.LikeContext context) {
-    Expression escape = null;
-    if (context.escape != null) {
-      escape = (Expression) visit(context.escape);
-    }
+    final Expression escape = null;
```
[the remark this answers is not in the document]
Comment (blueedgenick) [reply ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java]: fixed

File: ksql-parser/src/test/java/io/confluent/ksql/parser/rewrite/StatementRewriterTest.java [hunk 5]
```diff
@@ -22,6 +22,10 @@
 import static org.hamcrest.core.IsInstanceOf.instanceOf;
 import static org.hamcrest.core.IsNot.not;
 
+import org.junit.Assert;
```
[the remark this answers is not in the document]
Comment (blueedgenick) [reply ksql-parser/src/test/java/io/confluent/ksql/parser/rewrite/StatementRewriterTest.java]: not sure how that one crept in. fixed.

**Comment by blueedgenick** [comment]
thanks @big-andy-coates , figured you'd like this one! :-)

**Pull request merged**